### ISSUE TITLE
Imgur bugs

### ISSUE THREAD

**Issue by GitBoudewijn** [dsohler/oneclick-downloader#8]
Hey,

I decided to try out Imgur as well and found a couple of bugs:

1) On a Reddit page with Imgur links it gives a couple of false positives.

For example:
https://www.reddit.com/r/gentlemanboners/

Gives:
- http://i.imgur.com/uVyqtB4h.jpg
- **https://www.reddit.com/domain/i.imgur.com/**
- http://i.imgur.com/T9APKzW.jpg
- http://i.imgur.com/2zo3Nb8.jpg
- http://imgur.com/jNjarVz
- **https://www.reddit.com/domain/imgur.com/**
- http://i.imgur.com/t6optsC.jpg
- ...

2) Downloading from a Imgur album doesn't really work.

For example:
https://imgur.com/a/AxJNd?grid

Gives all kinds of links but not the images.

**Comment by dsohler** [comment]
Thanks for the report. The “Reddit and Imgur ecosystem” changed a lot since I was able to actually develop this extension the last time. I’ll check this and will update as soon as possible.

**Comment by dsohler** [comment]
False positives and not working site links are fixed in b39a65d.

Unfortunately the grid gallery download issue can't be fixed that easy because there are technically no links that could be matched and the injection that is used only works for links (it’s highly generalized and it is designed intentionally without the option to alter that behavior to keep it as simple and fast as possible).

You can circumvent that issue a little by switching to the non-grid view by removing the `?grid` from the URL, then run the extension on that site, select all links and manually unselect the links that are not images (there are a few but not too many)

**Comment by GitBoudewijn** [comment]
Yeah not being able to download from grid view is not that important, as long as it doesn't give false positives. I'll test it again when it's updated and close this issue if I don't find anything.

**Comment by dsohler** [comment]
I pushed the current development version to the store and it will be available as soon as Google synchronized it.

**Comment by GitBoudewijn** [comment]
The first issue is fixed but when downloading from an Imgur gallery it still gives a lot of false positives.

Here's someone who asked the same question:
https://stackoverflow.com/questions/26457485/how-would-i-detect-what-is-an-imgur-picture-link-and-what-isnt

Also maybe you shouldn't match album (**imgur.com/a/xxx**) links, since those contain multiple images. Right now when you try to download a link like that it only downloads a lower quality version of the first image in the album.

For example:
https://imgur.com/a/AxJNd

It only downloads **KZ6jX0Fg.jpg** (**KZ6jX0F.jpg** is the full size image).

**Comment by dsohler** [comment]
2014 it was quite easy to download imgur albums. It works best not using this extension on imgur albums :smile:

I'll check what I can do here to solve the issue.
